packages/actions-spec/index.d.ts

```diff
@@ -175,23 +182,37 @@ export interface ActionPostRequest<T = string> {
   data?: Record<keyof T, string | Array<string>>;
 }
 
+/** Type of action to determine client side handling */
+export type PostActionType = LinkedActionType;
+
 /**
- * Response body payload returned from the Action POST Request
+ * Response body payload returned from the Action POST Request if the action is a transaction
  */
-export interface ActionPostResponse<T extends ActionType = ActionType> {
-  /** base64 encoded serialized transaction */
+export interface TxResponse {
+  type: Extract<PostActionType, "tx">;
   transaction: string;
-  /** describes the nature of the transaction */
   message?: string;
   links?: {
-    /**
-     * The next action in a successive chain of actions to be obtained after
-     * the previous was successful.
-     */
     next: NextActionLink;
   };
 }
 
+/**
+ * Response body payload returned from the Action POST Request if the action is a POST request
+ */
+export interface PostResponse {
+  type: Extract<PostActionType, "post">;
+  message?: string;
+  links?: {
+    next: NextActionLink;
+  };
+}
+
+/**
+ * Response body payload returned from the Action POST Request
+ */
+export type ActionPostResponse = TxResponse | PostResponse;
```

Comment:
we can clean up the types here some and make them reusable by creating an generic interface for the different POST responses, I called it `ActionResponse`:

```suggestion
/** Type of action to determine client side handling */
export type PostActionType = LinkedActionType;

/**
 * Generic response from an Action API request
 */
export interface ActionResponse {
  type?: PostActionType;
  message?: string;
  links?: {
    next: NextActionLink;
  };
}

/**
 * Response body payload returned from the Action POST Request if the action is a transaction
 */
export interface TransactionResponse extends ActionResponse {
  type?: Extract<PostActionType, "transaction">;
  transaction: string;
}

/**
 * Response body payload returned from the Action POST Request if the action is a POST request
 */
export interface PostResponse extends ActionResponse {
  type: Extract<PostActionType, "post">;
}

/**
 * Response body payload returned from the Action POST Request
 */
export type ActionPostResponse = TransactionResponse | PostResponse;
```

Reply:
this suggested code block also avoids using the short name of `Tx` and uses `Transaction` instead

Reply:
This looks much better.